packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -908,7 +908,7 @@ export const ComboBoxMixin = (subclass) =>
         }
       } else {
         // Try to find an item which label matches the input value.
-        const items = [...(this._dropdownItems || []), this.selectedItem];
+        const items = [this.selectedItem, ...(this._dropdownItems || [])];
```

Comment:
When looking for an item that matches the input by label, check the currently selected item first

Reply:
Thanks Tomi, this fix was simpple than expected! :)